cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -2050,6 +2050,15 @@
     </description>
   </property>
 
+  <property>
+    <name>user.programs.emit.metrics</name>
```

Comment:
With the current naming convention, configurations that affect user program should be started with `app.program.`. Also, for boolean config, it should be ended with `.enabled`. Maybe call it `app.program.metrics.enabled`